superduper/backends/base/data_backend.py

```diff
@@ -131,6 +132,35 @@ def infer_schema(data: t.Mapping[str, t.Any], identifier: t.Optional[str] = None
         :return: The inferred schema
         """
 
+    def check_ready_ids(
+        self, query: Query, keys: t.List[str], ids: t.Optional[t.List[t.Any]] = None
+    ):
+        """Check if all the keys are ready in the ids.
+
+        :param query: The query object.
+        :param keys: The keys to check.
+        :param ids: The ids to check.
+        """
+        if ids:
+            query = query.select_using_ids(ids)
+        data = query.execute()
+        ready_ids = []
+        for select in data:
+            notfound = 0
+            for k in keys:
+                try:
+                    select[k]
+                except KeyError:
+                    notfound += 1
+            if notfound == 0:
+                ready_ids.append(select[query.primary_id])
+        if ids and len(ready_ids) != len(ids):
+            not_ready_ids = [id for id in ids if id not in ready_ids]
+            logging.info(f"IDs {not_ready_ids} do not ready.")
+            logging.debug(f"Ready IDs: {ready_ids}")
+            logging.debug(f"Not ready IDs: {not_ready_ids}")
+        return ready_ids
```

Comment:
Preserve the original logic as the default method, to be used as the default method for future new plugins.